### DISCUSSION TITLE
Building previous HammerDB versions in Docker

### DISCUSSION THREAD

**Discussion by DGonzalezVillal** [TPC-Council/HammerDB#769] [Q&A] [answered]
Hello!

So I'm currently trying to run HammerDB in docker, but I'm supposed to be using version 4.11. The steps I'm taking are, cloning the repo, checking out version "v4.11-docker", going to the docker folder and then building it.

But when I run the container in bash, it says I'm using version 4.12? Is this expected behavior? I'm I doing something wrong?

Please let me know!

**Comment by sm-shaw** [comment, marked as answer]
I've tested and can reproduce, it appears to be this line that pulls the latest base:
`=> [internal] load metadata for docker.io/tpcorg/hammerdb:base`

As a workaround, if you edit the Dockerfiles, so the main one has the first 2 lines as:

```
FROM  docker.io/tpcorg/hammerdb:v4.11 as oracle
FROM  docker.io/tpcorg/hammerdb:v4.11
```
and in the database specific directories it has this

`FROM docker.io/tpcorg/hammerdb:v4.11` rather than `hammerdb:base`
then in my testing, it correctly builds HammerDB v4.11 with all the libraries installed.

**Comment by DGonzalezVillal** [comment]
This got it working! Should I raise an issue about this?

**Reply by sm-shaw** [reply]
Yes, please do, and then we can ensure this gets fixed at the next version so when you checkout the tag with a specific docker version it should build at that version. The only potential drawback with the above approach is that if you want to build v4.11 for a specific database only with the above change then it is likely to build for all databases, however it does work. An ideal solution would be if we can do something like docker.io/tpcorg/hammerdb:v4.11: base (obviously this is not valid syntax) so we can just get the base at a specific version.
But yes, please raise an issue, and we can track fixing this in future with the above workaround in place for now.
